### PR TITLE
Initial commit

### PR DESCRIPTION
**Pull request by brianilles**

### CODE REVIEW

File: instagram/src/App.js [hunk 1]
```diff
@@ -1,14 +1,23 @@
 import React, { Component } from "react";
 import "./App.css";
+import PostPage from "./components/PostsContainer/PostsPage";
+import withAuthenticate from "./components/Authenication/withAuthenticate";
+import Login from "./components/Login/Login";
+
+// currying
```
Comment (secantsquared) [inline instagram/src/App.js:6]: this is exactly what this is - and is a very powerful concept from functional programming.

Which, I had it all wrong at first, I thought here was this OO based thing, but React when you get down to it is really purely functional in approach.

File: instagram/src/App.js [hunk 2]
```diff
@@ -1,14 +1,23 @@
...
+const ComponentFromWithAuthenticate = withAuthenticate(PostPage)(Login);
+
+export default class App extends Component {
+  constructor() {
+    super();
+    this.state = {};
```
Comment (secantsquared) [inline instagram/src/App.js]: Turns out with modern React you can save some time and just drop all of the constructor and super calls, and just start off with `state = {}`

File: instagram/src/components/Authenication/withAuthenticate.js [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+
+    componentDidMount() {
+      if (localStorage.getItem("user")) {
+        return this.setState({ loggedIn: true });
+      }
+      return this.setState({ loggedIn: false });
+    }
+
+    render() {
+      return this.state.loggedIn ? <PostPage /> : <LoginPage />;
```
Comment (secantsquared) [inline instagram/src/components/Authenication/withAuthenticate.js:20]: nice use of  a ternary to conditionally render here

File: instagram/src/components/CommentContainer/CommentSection.js [hunk 4]
```diff
@@ -0,0 +1,66 @@
...
+
+  render() {
+    return (
+      <div className="comment-container">
+        {this.state.commentsData.map((commentsDatum, index) => (
+          <CommentContainer key={index} comment={commentsDatum} />
+        ))}
+        <div className="time">
+          <p>
+            {moment(
```
Comment (secantsquared) [inline instagram/src/components/CommentContainer/CommentSection.js:61]: nice incorporation of momentjs

File: instagram/src/components/MediaContainer/MediaSection.js [hunk 5]
```diff
@@ -0,0 +1,12 @@
...
+// will have to change to class later
+const MediaSection = props => {
+  return (
+    <div className="img-container">
+      <img src={props.imageUrl} alt="media" />
+    </div>
+  );
+};
+
+export default MediaSection;
```
Comment (secantsquared) [inline instagram/src/components/MediaContainer/MediaSection.js]: Good call using this component - very organized.

File: instagram/src/components/PostsContainer/Post.js [hunk 6]
```diff
@@ -0,0 +1,38 @@
...
+class Post extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      likes: props.post.likes
+    };
+  }
+
+  addLike = () => {
+    this.setState({ likes: this.state.likes + 1 });
```
Comment (secantsquared) [inline instagram/src/components/PostsContainer/Post.js]: if you notice this gets a little funky you may wanna try
```suggestion
this.setState(prevState => {
     return {
                likes: prevState.likes + 1
    }
})

File: instagram/src/components/PostsContainer/PostsPage.js [hunk 7]
```diff
@@ -0,0 +1,43 @@
...
+  };
+
+  render() {
+    return (
+      <div>
+        <SearchBar handleSearch={this.handleSearch} />
+        <PostsContainer
+          postsData={
+            this.state.postsDataFiltered[0] == null
+              ? this.state.postsData
```
Comment (secantsquared) [inline instagram/src/components/PostsContainer/PostsPage.js:36]: Excellent uses of ternaries they will be your best friends here.

**Review by secantsquared — APPROVED** [review]
Good stuff Brian

**Comment by secantsquared** [comment]
@brilles I will allow you to manage merges due to the week long nature of this project

File: instagram/src/App.js (hunk 2 above)
Comment (brianilles) [reply instagram/src/App.js]: Thanks, fixed

File: instagram/src/components/PostsContainer/Post.js (hunk 6 above)
Comment (brianilles) [reply instagram/src/components/PostsContainer/Post.js]: Thanks, fixed. Your suggestion improves the readability for me

**Pull request merged**